core/src/main/java/io/undertow/io/BufferWritableOutputStream.java

```diff
@@ -34,6 +34,19 @@ public interface BufferWritableOutputStream {
 
     void write(final ByteBuffer byteBuffer) throws IOException;
 
+    /**
+     * Transfer the remaining content of the input FileChannel (from its current position, to end of file).
+     * @deprecated use {@link BufferWritableOutputStream#transferFrom(FileChannel, long, long)}.
+     */
+    @Deprecated
     void transferFrom(FileChannel source) throws IOException;
```

Comment:
I don't think we should deprecate this method but, instead, add the other method as an alternative.